active/gridsyntax/GridSyntaxSpec2.md

```diff
@@ -0,0 +1,368 @@
+          <RowDefinition Height="1*" MinHeight="100" />
+          <RowDefinition Height="Auto" />
+          <RowDefinition Height="25" />
+          <RowDefinition Height= "14" />
+          <RowDefinition Height="20" />
+    </Grid.RowDefinitions>
+</Grid>
+```
+
+
+# Remarks
+
+# API Details
+
+## ColumnDefinition API
+
+```cs
+[webhosthidden]
+
+// vvv
+[contentproperty("Width")] // Sets the ColumnDefinition's content property to the Width property.
+// ^^^
+
+runtimeclass ColumnDefinition : Windows.UI.Xaml.DependencyObject
+{
+    ColumnDefinition();
+    Windows.UI.Xaml.GridLength Width;
+    Double MaxWidth;
+    Double MinWidth;
+    Double ActualWidth{ get; };
+    static Windows.UI.Xaml.DependencyProperty WidthProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty MaxWidthProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty MinWidthProperty{ get; };
+};
+```
+
+## RowDefinition API
+
+```cs
+[webhosthidden]
+
+// vvv
+[contentproperty("Height")] // Sets the RowDefinition's content property to the Height property.
+// ^^^
+
+runtimeclass RowDefinition : Windows.UI.Xaml.DependencyObject
+{
+    RowDefinition();
+    Windows.UI.Xaml.GridLength Height;
+    Double MaxHeight;
+    Double MinHeight;
+    Double ActualHeight{ get; };
+    static Windows.UI.Xaml.DependencyProperty HeightProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty MaxHeightProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty MinHeightProperty{ get; };
+};
+```
+
+# Appendix
+
+## Alternatives
+
+## Syntax Considerations and Rationale
+In developing the syntax, the team considered developer familiarty and patterns as much as possible to ensure that the Grid learning curve could be decreased. One of the considerations was how similar the syntax should be to JSON, as that's a popular notation style that often addresses the same problems that Xaml does. 
+
+JSON defines their arrays as: ```"words" : ["apple", "banana", "cherry"]```
+
+Xaml defines a collection-typed property as: ```<Words="'Apple', 'Banana', 'Cherry'"/>```
+
+The syntax that we chose encloses strings individually in quotes, as JSON does, but it does so in a way that is more consistent with existing Xaml style and Xaml functions. An example of this syntax used elsewhere in Xaml is for the x:Bind syntax:
+
+```xml <TextBlock Text="{x:Bind VM.MyText, FallbackValue='Hello, world', Mode=TwoWay}" />```
+
+This uses a comma as a delimiter, but also allows strings to have commas within them. For these reasons, we chose to use the syntax in which a collection can be delimited by commas but have more complex values be encased within single quotes.
+
+
+## Release Details
+The xmlns will not be updated for this Xaml language feature addition, i.e. there will be no formal update to the Xaml language. Visual Studio, however, will have to be updated to accept the new syntax structure. This feature will be released with WinUI and not with the Xaml SDK.
```

Comment:
Is it legal to change the language without revving the language? This will lead to confusion if somebody takes some new-style markup and brings it to an old xaml compiler. The compiler will be totally happy with the version number but balk at the new synax.

Reply:
Discussion: In other words, do we need to update xmlns:x='http://schemas.microsoft.com/winfx/2006/xaml'?

For example, the namespace was updated for 2009.

We didn't update the language version number for other additive updates:
* x:Bind
* Conditional Xaml